driver-core/src/test/functional/com/mongodb/client/TestHelper.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright 2008-present MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.mongodb.client;
+
+import com.mongodb.lang.Nullable;
+
+import java.lang.reflect.Field;
+import java.util.Map;
+
+import static java.lang.System.getenv;
+
+public final class TestHelper {
+
+    public static void setEnvironmentVariable(final String name, @Nullable final String value) {
```

Comment:
Your idea, in discussion, was to use undocumented mechanism properties to pass this information in.

Will consider.

Reply:
We can also use a private `MongoClientSettings` method (via reflection) to pass test-specific object abstracting the driver code from the environment. Doing so will also not break when changing JDK, but may require test-specific module setup with the `--add-opens` CLI option for `java` if we ever modularize the driver (this setup will not be visible to users).

Reply:
I will omit. There are [incoming changes](https://jira.mongodb.org/browse/DRIVERS-2616) to the OIDC spec and prose tests. I propose that this be considered as part of that work, since I expect it might require the usage of this pattern for tests involving the cache anyway. (Altering variables in this way and adding undocumented/unsupported config to what is a public API both seem less than ideal.)